shuup/front/api/products.py

```diff
@@ -284,13 +316,23 @@ class Meta(CompleteShopProductSerializer.Meta):
 
 class FrontShopProductFilter(filters.BaseFilterBackend):
     """
-    Filter shop products by visible or listed products and not deleted ones
+    Filter shop products by visible or listed products and not deleted ones.
+    You can also filter by:
+        - shop - the ID of the shop
     """
     def filter_queryset(self, request, queryset, view):
-        return queryset.filter(
+        shop = request.query_params.get("shop")
+
+        shops_qs = queryset.filter(
             visibility__in=(ShopProductVisibility.LISTED, ShopProductVisibility.ALWAYS_VISIBLE),
             product__deleted=False
-        ).distinct()
+        )
+
+        # filter by shop
+        if shop:
+            shops_qs = shops_qs.filter(shop__id=shop)
```

Comment:
here instead do `.filter(shop_id=shop)` the `__` causes join if I have understood correctly.